Added PrestaShop version in bin/console

Codencode opened the pull request:
<!-----------------------------------------------------------------------------
Thank you for contributing to the PrestaShop project!

Please take the time to edit the "Answers" rows below with the necessary information.

Check out our contribution guidelines to find out how to complete it:
https://devdocs.prestashop-project.org/9/contribute/contribution-guidelines/#pull-requests

For type and category see:
https://devdocs.prestashop-project.org/9/contribute/contribution-guidelines/pull-requests/#type--category
------------------------------------------------------------------------------>

| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | It adds some information about PrestaShop to the "about" command in the console. For now, I added: Version, Debug mode, Cache.
| Type?             | improvement
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | In the console, run the command: `php bin/console about` and check if the additional information is displayed.
| UI Tests          | 
| Fixed issue or discussion?     | idea #39126 
| Related PRs       | 
| Sponsor company   | Codencode

Codencode committed:
Added PrestaShop version in bin/console

Codencode commented:
Considering that the `Symfony\Bundle\FrameworkBundle\Command\About` command cannot be extended, I reimplemented the command by copying the original one and created a dedicated method to add PrestaShop-related information.
This way, we have the information from Symfony's command plus PrestaShop's details.
The only drawback is that if Symfony's main command is modified, we lose those updates.
Could we consider creating a dedicated command for PrestaShop?

Progi1984 commented:
@Codencode Symfony solution : https://github.com/symfony/symfony/issues/46186#issuecomment-1119081971

Unregister & Register a new one

Codencode commented:
Hello @Progi1984, so do you think it's okay?

nicosomb reviewed (commented):
Great idea 🎉 You have some issues in CI.

jolelievre commented:
I was also wondering if you could include in this PR an override of https://github.com/symfony/console/blob/7.3/Command/ListCommand.php so that we can also have the PrestaShop version in the first line (as suggested in the issue)

```bash
$ bin/console 
PrestaShop 9.0.0 with Symfony 6.4.22 (env: dev, debug: true)
```

From what I found I think this can be done easily but extending `getLongVersion` in https://github.com/PrestaShop/PrestaShop/blob/b8ef1f164a4c327e1329aca5cc5395fb17e1cdcc/src/PrestaShopBundle/Console/PrestaShopApplication.php#L43

Codencode committed:
Refactor AboutCommand to extend Symfony's base command and avoid code duplication

Replaced the custom AboutCommand implementation with a subclass of Symfony's BaseAboutCommand
to prevent code duplication and ensure better maintainability as the upstream command evolves.

Codencode committed:
Override getLongVersion to display PrestaShop version in console output

Codencode commented:
Hello @jolelievre,
I think everything is clear now.

For now, I've created 2 commits:

- One to perform the refactor in order to revert to extending the Symfony\Bundle\FrameworkBundle\Command\AboutCommand.
- And one to add the PrestaShop version.

Now, what's left is the "move" to version 9.1.0, which I believe is done via rebase. I've never done it before, so I'll do my best.

Thanks for the suggestions!

Codencode committed:
phpstan baseline: skip final class extension error in AboutCommand

Codencode commented:
I also added the exception in the `phpstan-baseline.neon` file.

Quetzacoalt91 commented:
@Codencode Can you rebase to remove the uneeded commits ?

Codencode commented:
@Quetzacoalt91 rebase done

kpodemski commented on src/PrestaShopBundle/Command/AboutCommand.php:
```diff
@@ -0,0 +1,72 @@
+        $io = new SymfonyStyle($input, $output);
+
+        $rows = [
+            ['<info>Prestashop</>'],
```
```suggestion
            ['<info>PrestaShop</>'],
```

;-)

kpodemski commented on src/PrestaShopBundle/Command/AboutCommand.php:
```diff
@@ -0,0 +1,72 @@
+            new TableSeparator(),
+            ['Version', _PS_VERSION_],
+            ['Debug mode', _PS_MODE_DEV_ ? 'true' : 'false'],
+            ['Cache', $this->configuration->get('PS_SMARTY_CACHE') ? 'true' : 'false'],
```
```suggestion
            ['Smarty Cache', $this->configuration->get('PS_SMARTY_CACHE') ? 'true' : 'false'],
```

Remember that there's a Smarty cache and an actual Cache section, where you can choose between Memcache, etc.

Codencode replied:
You're absolutely right!

Codencode committed:
Update labels

Codencode commented:
@kpodemski, changes applied.

Codencode committed:
Correct tag closing in AboutCommand.php

SiraDIOP reviewed (commented):
Qa by Dev

PululuK commented on src/PrestaShopBundle/Command/AboutCommand.php:
```diff
@@ -0,0 +1,72 @@
+
+        $io->table([], $rows);
+
+        return 0;
```
```suggestion
        return self::SUCCESS;
```

Codencode replied:
Done, thank you!

Codencode committed:
Added self::SUCCESS in AboutCommand.php"

kpodemski commented:
@PrestaShop/committers ready to be tested by a Dev :)

Codencode commented:
@kpodemski, are these 2 screenshots enough? And most importantly, can I do it since I'm the one who created the PR?
### Without fix
![without-fix](https://github.com/user-attachments/assets/2a48ba67-f721-456c-9a18-877f09f5f6bc)

### With fix
![with-fix](https://github.com/user-attachments/assets/06168551-b514-4605-a85e-908576586363)

Codencode commented:
@kpodemski 
As I imagined, thank you!

Touxten commented:
Hey @Codencode i fork your branch when i launch `php bin/console about` i have an issue it's possible i do a mistake ?

<img width="2196" height="931" alt="Capture d’écran du 2025-08-20 15-04-33" src="https://github.com/user-attachments/assets/53c4f3aa-e5c2-4135-b429-125077576e97" />
